Issues/95 - Add Run 2.1i DR1 Object Table tutorial

wmwv opened the pull request:
Adds a tutorial stepping through the Run 2.1i DR1 Object Table.

joezuntz commented on validation/validate_dc2_run2.1i_object_table.ipynb:
```diff
@@ -0,0 +1,713 @@
+   "outputs": [],
+   "source": [
+    "for filt in filters:\n",
+    "    df['e_{}'.format(filt)], df['e1_{}'.format(filt)], df['e2_{}'.format(filt)] = \\\n",
```
When I get to this cell on jupyter.nersc.gov the kernel silently dies.  I think it's running out of memory.

So I can't really review beyond here.

wmwv replied:
Can you try with the restricted one-tract catalog?
I've added it as an option in the first cell to allow someone to run the Notebook in a finite amount of time.

I will also add a completed run to the `rendered` branch.

wmwv commented on validation/validate_dc2_run2.1i_object_table.ipynb:
```diff
@@ -0,0 +1,713 @@
+    "https://docs.google.com/document/d/18nNVImxGioQ3tcLFMRr67G_jpOzCIOdar9bjqChueQg/view\n",
+    "https://github.com/LSSTDESC/DC2_visitList/blob/master/DC2visitGen/notebooks/DC2_Run2_regionCoords_WFD.ipynb\n",
+    "\n",
+    "| Location        | RA (degrees) | Dec (degrees) | RA (degrees) | Dec (degrees) |\n",
```
I fixed this by starting each table row with a "|".  I previously only had a few rows starting with a "|" and the rest didn't.  The GitHub viewer apparently didn't like that.  Thanks for catching.

wmwv commented on validation/validate_dc2_run2.1i_object_table.ipynb:
```diff
@@ -0,0 +1,713 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plot_ra_dec(df)"
```
The goal is for run Notebooks to be available in the `rendered` branch.
I will make sure that this works and exists in that branch.

wmwv commented on validation/validate_dc2_run2.1i_object_table.ipynb:
```diff
@@ -0,0 +1,713 @@
+    "    ra_left_side_slope = ra_left_side_delta / dec_size\n",
+    "    ra_right_side_slope = ra_right_side_delta / dec_size\n",
+    "\n",
+    "    inside_ra = (corners[2][0] + ra_left_side_slope * (df['dec'] - corners[2][1]) < df['ra']) & \\\n",
```
Oh, thanks!  Clearly a residual from promoting to a function without actually changing the variables.

sschmidt23 commented on validation/validate_dc2_run2.1i_object_table.ipynb:
```diff
@@ -0,0 +1,733 @@
+    "* We're missing the DDF region, which was specifically not included in this DR1 coadd.\n",
+    "* We're missing a bit of the South-West (lower-left, \"North Up, East Left\") corner.\n",
+    "* There are also a few patches that failed within the main region.\n",
+    "* There is an overall gradient N/S in object density, because we're plotting in rectilinear RA, Dec bins, which means that bins at the bottom in RA cover less area than those at the top.\n",
```
If you plotted only stars/extendedness==0 would you see a gradient due to Galactic latitude, or is this too small a range to spot that?  If it is visible, might be a nice additional plot to make

wmwv replied:
No, too small a range at too high above the Galactic plane to see a trend.

sschmidt23 commented on validation/validate_dc2_run2.1i_object_table.ipynb:
```diff
@@ -0,0 +1,733 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.hexbin(good['mag_i'], good['extendedness'],\n",
```
The plot shows that extendedness is a binary quantity, is it worth it to also show mag_psf - mag_cModel vs mag_cModel (i.e. Figure 16 from the Bosch et al HSC paper) to remind people that extendedness, while binary, is sensitive to an ad hoc threshold choice in the delta_mag?  That might be more informative when paired with the binary extendedness plot.

wmwv replied:
Good suggestions.  Done.

RobertLuptonTheGood commented:
We (== DM) have SVM (?) code to do a better job of s/g separation.  It's been used trained on HSC data (using HST COSMOS imaging as space truth, although HSC is too deep), and we could probably run in on DESC sims too.

			R

wmwv commented on validation/validate_dc2_run2.1i_object_table.ipynb:
```diff
@@ -0,0 +1,713 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def inside_trapezoid(corners, ra, dec):\n",
```
⚪️ ▪️

wmwv commented:
Updates now pushed:
* Default to `*_tract4850`.  This takes ~10 seconds to load.
* Added subsampling logic to make easy subsamples when loading full DC2.  Scanning the full set of data files still takes ~5 minutes, but the reduced memory footprint means the JupyterHub node doesn't spend all of its time swapping.
* Added Mandelbaum N(mag_i) HSC comparison for galaxies.

wmwv commented:
Thanks @joezuntz  @sschmidt23 for your reviews.

Even in the absence of formal approval from the reviewers, I'm taking the liberty of merging this to get this Notebook in front of the wider collaboration.
